ipn/ipnlocal/profiles.go

```diff
@@ -206,7 +207,12 @@ func init() {
 // SetPrefs sets the current profile's prefs to the provided value.
 // It also saves the prefs to the StateStore. It stores a copy of the
 // provided prefs, which may be accessed via CurrentPrefs.
-func (pm *profileManager) SetPrefs(prefsIn ipn.PrefsView) error {
+//
+// If nm is provided non-nil, it will be used to enrich the profile
+// with the current self node's MagicDNS suffix. The MagicDNS suffix
+// cannot be pulled from prefsIn directly because it is not saved on
+// ipn.Prefs (since it's not a field that is configurable by nodes).
+func (pm *profileManager) SetPrefs(prefsIn ipn.PrefsView, nm *netmap.NetworkMap) error {
```

Comment:
Can you have this just take the tailnetName as the input? NetMap seems pretty heavyweight

Reply:
updated!